File: src/main/java/com/esprit/models/Programme.java
package com.esprit.models;

public class Programme {

    private int id;
    private double prix;
    private String description;
    private int coach_id;
    private String lieu;

    public Programme( int id,double prix, String description, int coach_id,String lieu) {

        this.id=id;
        this.prix = prix;
        this.description = description;
        this.coach_id = coach_id;
        this.lieu=lieu;
    }
    // ✅ Constructeur qui accepte java.util.Date et les convertit en LocalDate
    public Programme( double prix, String description, int coach_id,String lieu) {


        this.prix = prix;
        this.description = description;
        this.coach_id = coach_id;
        this.lieu=lieu;
    }




    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }



    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
    public int getCoach_id() {
        return coach_id;
    }
    public void setCoach_id(int coach_id) {
        this.coach_id = coach_id;
    }
    public String getLieu(){return lieu;}

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    @Override
    public String toString() {
        return "Programme{" +
                "id=" + id +

                ", prix=" + prix +
                ", description='" + description + '\'' +
                ", coach_id=" + coach_id +
                ", lieu=" + lieu +
                '}';
    }


}

File: src/Entity/Programme.php
<?php

namespace App\Entity;

use App\Repository\ProgrammeRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
#[ORM\Entity(repositoryClass: ProgrammeRepository::class)]
class Programme
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Le prix ne doit pas être vide.")]
    #[Assert\Positive(message: "Le prix doit être supérieur à zéro.")]
    private ?float $prix = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "La description est obligatoire.")]
    #[Assert\Length(
        min: 5,
        minMessage: "La description doit faire au moins 5 caractères."
    )]
    private ?string $description = null;
    #[ORM\ManyToOne(inversedBy: 'programmes')]
    #[ORM\JoinColumn(nullable: false, name: 'coach_id', referencedColumnName: 'id',onDelete: 'CASCADE')]
    #[Assert\NotNull(message: "Le coach est requis.")]
    private ?Coach $coach_id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le lieu ne doit pas être vide.")]
    private ?string $lieu = null;

    /**
     * @var Collection<int, Application>
     */
    #[ORM\OneToMany(targetEntity: Application::class, mappedBy: 'idProgramme', cascade: ['remove'], orphanRemoval: true)]
    private Collection $applications;
    

    /**
     * @var Collection<int, Seance>
     */
    #[ORM\OneToMany(targetEntity: Seance::class, mappedBy: 'programme_id', cascade: ['remove'], orphanRemoval: true)]
    private Collection $seances;
    
    public function __construct()
    {
        $this->applications = new ArrayCollection();
        $this->seances = new ArrayCollection();
    }


    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): static
    {
        $this->id = $id;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): static
    {
        $this->prix = $prix;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }
    public function getCoachId(): ?Coach
    {
        return $this->coach_id;
    }

    public function setCoachId(?Coach $coach_id): static
    {
        $this->coach_id = $coach_id;

        return $this;
    }
    

    public function getLieu(): ?string
    {
        return $this->lieu;
    }

    public function setLieu(string $lieu): static
    {
        $this->lieu = $lieu;

        return $this;
    }

    /**
     * @return Collection<int, Application>
     */
    public function getApplications(): Collection
    {
        return $this->applications;
    }

    public function addApplication(Application $application): static
    {
        if (!$this->applications->contains($application)) {
            $this->applications->add($application);
            $application->setIdProgramme($this);
        }

        return $this;
    }

    public function removeApplication(Application $application): static
    {
        if ($this->applications->removeElement($application)) {
            // set the owning side to null (unless already changed)
            if ($application->getIdProgramme() === $this) {
                $application->setIdProgramme(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection<int, Seance>
     */
    public function getSeances(): Collection
    {
        return $this->seances;
    }

    public function addSeance(Seance $seance): static
    {
        if (!$this->seances->contains($seance)) {
            $this->seances->add($seance);
            $seance->setProgrammeId($this);
        }

        return $this;
    }

    public function removeSeance(Seance $seance): static
    {
        if ($this->seances->removeElement($seance)) {
            // set the owning side to null (unless already changed)
            if ($seance->getProgrammeId() === $this) {
                $seance->setProgrammeId(null);
            }
        }

        return $this;
    }

    
}
